### PR TITLE
Image Upload - Terraform Admin S3 Bucket

### PR DESCRIPTION
**Pull request by anmho**
Terraform Infrastructure for image uploads
Initial database schema and initial migration setup

Eventually, the user flow for image upload will look something like this

1) Client calls GetImageUploadURL()
2) Server will return a presigned upload url to the client.
3) We will link events to image using the s3 file key. Default policy will be public readonly.

Note: GetImageUploadURL will have an aggressive rate limit: 5 times per minute to prevent abuse.

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -9,15 +7,29 @@ deps:
...
 watch:
-	@air --build.cmd "make build" --build.bin "./api/bin/api"
+	@air --build.cmd "make api" --build.bin "./api/bin/api"
 
 .PHONY: gen
 gen:
 	@buf generate
 
+
+.PHONY: init-tf
```
Comment (anmho) [inline Makefile:27]: Terraform shortcut commands

File: .gitignore [hunk 2]
```diff
@@ -1,3 +1,4 @@
 /tmp/
 /api/bin/
-.idea
\ No newline at end of file
+.idea
+**/.terraform/
```
Comment (anmho) [inline .gitignore:4]: Contains terraform secrets

File: api/docker-compose.yaml [hunk 3]
```diff
@@ -0,0 +1,22 @@
+version: '3.9'
+
+services:
+  db:
+    image: postgres
+    restart: always
+    # set shared memory limit when using docker-compose
+    shm_size: 128mb
+    # or set shared memory limit when deploy via swarm stack
```
Comment (anmho) [inline api/docker-compose.yaml:9]: docker-compose local development postgres

File: Makefile [hunk 4]
```diff
@@ -9,15 +7,29 @@ deps:
 	@go install "github.com/air-verse/air@latest"
 
 
-.PHONY: build
-build:
+.PHONY: api
```
Comment (anmho) [inline Makefile:12]: Renaming for clarity

File: api/inspect.sh [hunk 5]
```diff
@@ -0,0 +1,4 @@
+
+atlas schema inspect -u "postgres://admin:admin@localhost:5433/happened_db?sslmode=disable" --format '{{ sql . }}' > schema.sql
```
Comment (anmho) [inline api/inspect.sh:2]: Introspects database and returns current schema

File: api/migrate.sh [hunk 6]
```diff
@@ -0,0 +1,3 @@
+#!/usr/bin/env bash
+
+atlas schema apply --url "postgres://admin:admin@localhost:5433/happened_db?sslmode=disable" --to "file://sql/schema.sql" --dev-url "docker://postgres/15"
```
Comment (anmho) [inline api/migrate.sh:3]: Applies the specified schema file to the database, similar to Terraform.

File: protos/v1/happened_messages.proto [hunk 7]
```diff
@@ -0,0 +1,29 @@
+syntax = "proto3";
+
+package happened_messages;
+
+option go_package = "happenedapi/gen/protos/v1;happenedv1";
+
+message GreetRequest {
```
Comment (anmho) [inline protos/v1/happened_messages.proto:7]: Moved out of service package to separate concerns.

File: Makefile [hunk 8]
```diff
@@ -9,15 +7,29 @@ deps:
...
+init-tf:
+	@terraform -chdir=./terraform init
+.PHONY: tf
+tf:
+	@terraform -chdir=./terraform apply
+
+.PHONY: destroy
+destroy:
+	@terraform -chdir=./terraform destroy
+
```
Comment (Jish2) [inline Makefile:38]: can you add terraform cli to deps?

**Comment by Jish2** [comment]
in the pr could you outline (high level) what our workflows with the s3 bucket might look like?

File: Makefile (hunk 8 above)
Comment (anmho) [reply Makefile:38]: Good point, will add

**Review by Jish2 — APPROVED** [review]
😎

**Pull request merged**